solc/CommandLineInterface.cpp

```diff
@@ -777,7 +777,10 @@ bool CommandLineInterface::processInput()
 	}
 
 	m_compiler.reset(new CompilerStack(fileReader));
+
 	auto scannerFromSourceName = [&](string const& _sourceName) -> solidity::Scanner const& { return m_compiler->scanner(_sourceName); };
+	SourceReferenceFormatter formatter(cerr, scannerFromSourceName);
```

Comment:
The type of a lambda function is different from `std::function`, you can only implicitly convert into `std::function`. This means since `auto` is used above, the type of `scannerFromSourceNam` is not `std::function`. Since `formatter` takes a `const&`, the type is converted on the fly, creating a temporary copy of `scannerFromSourceName`, converted to `std::function`. Since `SourceReferenceFormatter` (used to) store a `const&`, it stores a `const&` to this temporary which is destroyed after this line and thus invalid. I changed it to storing a value instead of a reference.

Reply:
That’s awesome 👏 . Thanks for taking the time to review this Chris!

I should definitely improve my understanding of that part of the language.

Reply:
Yeah, that's the problem with C++: It does not warn you about ownership problems.